doc/development/tutorials/autodoc_ext.rst

```diff
@@ -0,0 +1,120 @@
+.. _autodoc_ext_tutorial:
+
+Developing autodoc extension for IntEnum
+========================================
+
+The objective of this tutorial is to create an extension that adds
+support for new type for autodoc. This autodoc extension will format
+the ``IntEnum`` class from Python standard library. (module ``enum``)
+
+Overview
+--------
+
+We want the extension that will create auto-documentation for IntEnum.
+``IntEnum`` is the integer enum class from standard library ``enum`` module.
+
+Currently this class has no special auto documentation behavior.
+
+We want to add following to autodoc:
+
+* A new ``autointenum`` directive that will document the ``IntEnum`` class.
+* The generated documentation will have all the enum possible values
+  with names.
+* The ``autointenum`` directive will have an option ``:hex:`` which will
+  cause the integers be printed in hexadecimal form.
+
+
+Prerequisites
+-------------
+
+We need the same setup as in :doc:`the previous extensions <todo>`. This time,
+we will be putting out extension in a file called :file:`autodoc_intenum.py`.
+The :file:`my_enums.py` will contain the sample enums we will document.
+
+Here is an example of the folder structure you might obtain:
+
+.. code-block:: text
+
+      └── source
+          ├── _ext
+          │   └── autodoc_intenum.py
+          ├── conf.py
+          ├── index.rst
+          └── my_enums.py
+
+
+Writing the extension
+---------------------
+
+Start with ``setup`` function for the extension.
+
+.. literalinclude:: examples/autodoc_intenum.py
+   :language: python
+   :linenos:
+   :pyobject: setup
+
+
+The :meth:`~Sphinx.setup_extension` method will pull the autodoc extension
+because our new extension depends on autodoc. :meth:`~Sphinx.add_directive`
+is the method that registers our new auto documenter class.
+
+
+This is the definition of the auto-documenter class:
+
+.. literalinclude:: examples/autodoc_intenum.py
+   :language: python
+   :linenos:
+   :lines: 1-48
+
+Important attributes of the new class:
+
+* **objtype** - this attribute determines the ``auto`` directive name. In
+  this case the auto directive will be ``autointenum``.
+* **directivetype** - this attribute sets the generated directive name. In
+  this example the generated directive will be ``.. :py:class::``.
+* **priority** - the largest the number the higher is the priority. We want our
+  documenter be higher priority than the parent.
+* **option_spec** - option specifications. We copy the parent class options and
+  add a new option *hex*.
```

Comment:
It would be better to use a "definition list" instead. What do you think?
https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#definition-lists

Reply:
Sure